### PR TITLE
Create hud

### PR DESCRIPTION
**Pull request by KeyraWebb**
Created a HUD for the player to clearly see their money as well as some controls

### CODE REVIEW

File: Assets/Scripts/HUDManager.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+public class HUDManager : MonoBehaviour
+{
+    public static HUDManager Instance;
+    [SerializeField]
+    private GameObject hudObject;
+    private VisualElement hud;
+    private VisualElement moneyContainer;
+    private Label moneyDisplay;
+
+    // Start is called before the first frame update
```
Comment (lolyay270) [inline Assets/Scripts/HUDManager.cs:16]: This comment is incorrect for the method below it

Comment (lolyay270) [inline Assets/Scripts/HUDManager.cs:16]: This comment is incorrect for the method below it

File: Assets/Scripts/HUDManager.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    void Awake()
+    {
+        Instance = this;
+        hud = hudObject.GetComponent<UIDocument>().rootVisualElement;
+    }
+
+    private void OnEnable(){
```
Comment (lolyay270) [inline Assets/Scripts/HUDManager.cs:23]: There is inconsistent use of curly braces `{ }`, here they are in JS standard, but in `Awake()` the braces are in C# standard

File: Assets/UI/hud.uxml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<ui:UXML xmlns:ui="UnityEngine.UIElements" xmlns:uie="UnityEditor.UIElements" editor-extension-mode="False">
+    <Style src="project://database/Assets/UI/hud.uss?fileID=7433441132597879392&amp;guid=a2c53dd0d6d334a4191411c6f7313165&amp;type=3#hud" />
+    <ui:VisualElement name="moneyContainer" class="money-display hud-container" style="flex-grow: 1;">
+        <ui:Label name="moneyDisplay" text="$0" />
+    </ui:VisualElement>
+    <ui:VisualElement name="controlHints" class="hud-container control-display" style="flex-grow: 1;">
+        <ui:Label text="Shop: Q" name="shopControl" />
+        <ui:Label text="Inventory: Tab" name="inventoryControl" select-word-by-double-click="false" />
```
Comment (lolyay270) [inline Assets/UI/hud.uxml:8]: does `select-word-by-double-click="false"` need to be on this HUD element, since we never are able to click it?

**Review by lolyay270 — CHANGES_REQUESTED** [review]
The HUD is simple and effective. It works perfectly at most screen sizes.

Although I have noticed that if a user has a very wide monitor, the HUD elements are rather larger. I was wondering if it is possible to scale them based on which screen axis is larger.

I have left some comments in specific code lines for tweaks and optimisations

I appreciate the background tint, it allows the users to see the contents no matter what colour/brightness of game is behind it

File: Assets/UI/hud.uxml (hunk 3 above)
Comment (KeyraWebb) [reply Assets/UI/hud.uxml:8]: I didn't write this code, it was automatic. I'm going to double check that it's not gonna cause problems

**Comment by KeyraWebb** [comment]
Re-review needed, requested changes made.

**Review by lolyay270 — APPROVED** [review]
All changes made work, and fulfill the changes requested in the previous review

**Pull request merged**